x/amm/types/key_batch_txs.go

```diff
@@ -0,0 +1,64 @@
+package types
+
+import (
+	"encoding/binary"
+	fmt "fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ binary.ByteOrder
+
+const (
+	TLastSwapRequestIndex  = "last-swap-request-index"
+	TSwapExactAmountInKey  = "batch/swap-exact-amount-in"
+	TSwapExactAmountOutKey = "batch/swap-exact-amount-out"
+)
+
+func TKeyPrefixSwapExactAmountInPrefix(m *MsgSwapExactAmountIn) []byte {
+	prefix := []byte(m.TokenIn.Denom + "/")
+	routeKeys := []string{}
+	for _, route := range m.Routes[:1] {
+		routeKeys = append(routeKeys, fmt.Sprintf("%d/%s", route.PoolId, route.TokenOutDenom))
+	}
+	prefix = append(prefix, []byte(strings.Join(routeKeys, "/"))...)
+	return prefix
+}
+
+func FirstPoolIdFromSwapExactAmountIn(m *MsgSwapExactAmountIn) uint64 {
+	for _, route := range m.Routes {
+		return route.PoolId
```

Comment:
@jelysn this code can be confusing for readability, we are using a loop while we always stop at the first iteration, consider accessing directly to the first item of the list if available otherwise return 0